lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -139,6 +139,17 @@ def exists?
       @instance_data.present?
     end
 
+    def current_generation
+      return -1 unless exists? # must be different default than observed_generation
+      @instance_data["metadata"]["generation"]
+    end
+
+    def observed_generation
+      return -2 unless exists?
+      # populating this is a best practice, but not all controllers actually do it
```

Comment:
Out of curiosity, do you have an example of a controller that doesn't populate it?